### PR TITLE
Fixes #16772: Add an upgrade scenario in rtf

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/16772

### CODE REVIEW

File: rtf [hunk 1]
```diff
@@ -613,6 +613,11 @@ class Platform:
     self.hosts = {}
     self.plugins = {}
     self.provider = "virtualbox"
+
+    # Initialized at setup time
+    script_dir = os.path.dirname(os.path.realpath(__file__))
+    client_path = script_dir + "/rudder-api-client"
```
Comment (peckpeck) [inline rtf]: why not directly put this ?
self.client_path = script_dir + "/rudder-api-client"

File: scenario/create_user.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+"""
+Scenario: base
```
Comment (peckpeck) [inline scenario/create_user.py]: Name doesn't match

File: scenario/upgrade.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import scenario.create_user as create_user
+
+# Test base rudder and accept nodes
+accept_nodes.execute()
+create_user.execute("test_user1", "3188d0c4-db97-4a8d-bb05-2e8061b991b8")
+# Upgrade
+test_shell_on("server", '/usr/local/bin/rudder-setup upgrade-server 6.0-nightly', error_mode=Err.CONTINUE, live_output=True)
+# Update API token and re-test the webapp
+(rudder_url, token) = scenario.platform.api_connection_info()
+os.environ['RUDDER_TOKEN'] = token
```
Comment (peckpeck) [inline scenario/upgrade.py]: This should not work, os.environ is read only, writing to it doesn't affect environment

Comment (peckpeck) [reply scenario/upgrade.py]: Ok it seems to be used as the default for sub process at least

Comment (peckpeck) [inline scenario/upgrade.py]: Many other variables set by rtf are not set here, I think that you should put the code in common

**Comment by Fdall** [comment]
PR updated with a new commit

**Pull request merged**